pkg/azuredisk/nodeserver.go

```diff
@@ -335,25 +340,28 @@ func (d *Driver) NodeGetInfo(ctx context.Context, req *csi.NodeGetInfoRequest) (
 			zone, zoneError = d.cloud.GetZone(ctx)
```

Comment:
/hold

if `d.supportZone` is true and we're in a Windows node that doesn't have access to the metadata server then we'll always attempt to make this request (which times out but after some time), this means that the NodeGetInfo call will not succeed after some time has passed.

In the patch I made sure that if we get info from the API server that we no longer make requests to the metadata server, therefore NodeGetInfo succeeded with a single request

Setting `d.supportZone` to false through a cli flag would skip this block but the impl above is already trying to fetch info from the metadata server anyways

Is there a case where we would need `d.supportZone` given that the node component can get it from the kube API server?

Reply:
from the logs, I could see after timeout, it could proceed in `NodeGetInfo`, since `NodeGetInfo` is one time operation, I would suggest always get zone from instance metadata first (it's always accurate), and if failed with timeout, then get zone info from node labels.

```
azuredisk-plugin I0218 00:03:36.911998    9572 utils.go:77] GRPC call: /csi.v1.Node/NodeGetInfo
azuredisk-plugin I0218 00:03:36.911998    9572 utils.go:78] GRPC request: {}
azuredisk-plugin W0218 00:03:57.924669    9572 nodeserver.go:334] get instance type(w4abj7whz000000) failed with: Get "http://169.254.169.254/metadata/instance?api-version=2019-03-11&format=json": dial tcp 169.254.169.254:80: conne
ctex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
azuredisk-plugin W0218 00:04:18.951384    9572 nodeserver.go:358] get zone(w4abj7whz000000) failed with: Get "http://169.254.169.254/metadata/instance?api-version=2019-03-11&format=json": dial tcp 169.254.169.254:80: connectex: A c
onnection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
azuredisk-plugin I0218 00:04:18.951384    9572 nodeserver.go:387] not found a matching size in getMaxDataDiskCount, VM Size: , use default volume limit: 16
```

Reply:
Please look at https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1200#discussion_r818392423, the concern is the blocking call, in the example it took around 40s to timeout (I guess 20s per blocking call) before continuing

Reply:
already added `get-node-info-from-labels` flag, if it's set, then only get node info from node labels. PTAL.